test/test_optim.py

```diff
@@ -2665,5 +2665,26 @@ def test_bn_update_eval_momentum(self):
 
 instantiate_parametrized_tests(TestLRScheduler)
 
+def _diff_fn(p, grad, opt_differentiable_state, opt):
+    p = p.clone()
+    p.grad = grad
+    opt.update_parameters([p])
+    opt.state.update(opt_differentiable_state)
+    opt.step()
+    return p
+
+
+class TestDifferentiableOptimizer(TestCase):
+
+    def test_sgd(self):
+        p = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        grad = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        mbuff = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        state = {'momentum_buffer': mbuff}
+        opt = torch.optim.SGD([p], lr=0.9, differentiable=True)
```

Comment:
moving this inside _diff_fn should remove the need for the param swapping function.

Reply:
done!